### ISSUE TITLE
Generated JSON format is not correct.

### ISSUE THREAD

**Issue by FrankBorrero** [oeg-upm/gtfs-bench#117]
When generating JSON files with the benchmark,  they are created with a wrong format. Some brackets are missing.

Ex:

Generated JSON :

{"agency_id": "00000000000000000001", "agency_name": "00000000000000000001", "agency_url": "00000000000000000001", "agency_timezone": "00000000000000000001", "agency_lang": "00000000000
000000001", "agency_phone": "00000000000000000001", "agency_fare_url": "00000000000000000001"}

Proper JSON format:

**[**{"agency_id": "CRTM", "agency_name": "Consorcio Regional de Transportes de Madrid", "agency_url": "http://www.crtm.es", "agency_timezone": "Europe/Madrid", "agency_lang": "es", "agency_phone": "", "agency_fare_url": ""}**]**

**Comment by daniel-dona** [comment]
I don't know if this is really a bug or not, maybe @dachafra could also say something about, but this is the story...

VIG generate data as CSV files, that's not a problem but we also want some more formats (JSON and XML), so we use third-party tools to do the conversion. 

The JSON files are the result of using csvjson (https://csvkit.readthedocs.io/en/latest/scripts/csvjson.html), and this tool has a problem, a memory problem. Reading a 5GB CSV file to generate a JSON file is not a good idea, the only wait this tool provide to process data in a memory-efficient way is to generate a JSON stream (--stream option) (https://github.com/oeg-upm/gtfs-bench/blob/47c384efaf95ef5c6d36aa84c99e2b36c9fc2d1d/composer/app.py#L518), not a JSON array.

The only difference is the one you already pointed, the brackets and the comma-separated objects instead of a new line per object.

The important question about this is, I think, the programs that will consume this JSON can use a JSON stream? If not, we can implement a better csvjson (or maybe send a PR to their repository).

**Comment by dachafra** [comment]
changing the iterator from '$.[*]' to '$' in RMLMapper works. @FrankBorrero will test it over SDM-RDFizer and RocketRML. In the case it works we will change it over the generator in the mappings for json

**Comment by dachafra** [comment]
we will solve it changing the csvkit to our own script for transforming large csv file to json arrays. Additionally, we will open an issue in csvkit repo describing current issue we have using their library

**Comment by daniel-dona** [comment]
I created a little tool to do the correct JSON format generation ***and*** using a streaming/low-memory approach, this should be fixed now.

https://github.com/daniel-dona/csv2all / https://pypi.org/project/csv2all/

Also, I noticed that all data types are ALL strings... ¿can this cause any problem later? I can use type inference in Python, but we were already not using it in the old JSON converter.

And, also, I have the idea to replace the current CSV to XML converter and implement that too (in this tool), maybe we can make the conversion faster...
